src/python/pants/backend/terraform/dependency_inference.py

```diff
@@ -143,8 +147,44 @@ async def infer_terraform_module_dependencies(
     return InferredDependencies(terraform_module_addresses)
 
 
+@dataclass(frozen=True)
+class GetTerraformDependenciesRequest:
+    source_files: SourceFiles
+    directories: Tuple[str, ...]
+
+
+@dataclass(frozen=True)
+class TerraformDependencies:
+    fetched_deps: Tuple[Tuple[str, Digest], ...]
+
+
+@rule
+async def get_terraform_providers(
+    req: GetTerraformDependenciesRequest,
+) -> TerraformDependencies:
+    fetched_deps = await MultiGet(
+        Get(
+            FallibleProcessResult,
+            TerraformProcess(
+                args=("init",),
+                input_digest=req.source_files.snapshot.digest,
+                output_files=(".terraform.lock.hcl",),
+                output_directories=(".terraform",),
```

Comment:
`terraform` will write these relative to the `chdir`, but Pants doesn't know about `chdir` so won't it try to capture them relative to the sandbox root, and then not find them? So don't you need to prefix the `directory` here?

Reply:
They're prefixed inside of TerraformProcess. The logic for me was that terraform interprets all of the other file paths as relative to the chdir. Your comment reminded me that there's a better way than banging it together with pathlib, I'll look into that